packages/i18nify-js/src/modules/currency/formatNumber.ts

```diff
@@ -16,21 +17,32 @@ const formatNumber = (
       `Parameter 'amount' is not a number. typeof amount: ${typeof amount}`,
     );
 
-  let formattedAmount = '';
-
   try {
-    formattedAmount = getIntlInstanceWithOptions(options).format(
+    const formattedAmount = getIntlInstanceWithOptions(options).formatToParts(
       Number(amount),
     );
+
+    const parts: Array<{ type: string; value: unknown }> = formattedAmount;
+    const intlOptions = options?.intlOptions ? { ...options.intlOptions } : {};
+    const currencyCode = (options?.currency || intlOptions.currency) as string;
+
+    parts.forEach((p: { type: string; value: any }) => {
+      if (p.type === 'currency' && currencyCode in INTL_MAPPING) {
```

Comment:
If this is just for 1 type, we don't have to run this in forEach. 
Run it in a traditional for loop and break as soon as you find the currency type and make the substitution.